kubelet/tests/test_kubelet.py

```diff
@@ -53,23 +52,21 @@
 ]
 
 EXPECTED_METRICS_PROMETHEUS = [
+    # cadvisor metrics
     'kubernetes.memory.usage_pct',
     'kubernetes.network.rx_dropped',
     'kubernetes.network.rx_errors',
     'kubernetes.network.tx_dropped',
     'kubernetes.network.tx_errors',
     'kubernetes.io.write_bytes',
-    'kubernetes.io.read_bytes'
-]
-
-Label = namedtuple('Label', 'name value')
-
+    'kubernetes.io.read_bytes',
 
-class MockMetric(object):
-    def __init__(self, name, labels=None, value=None):
-        self.name = name
-        self.label = labels if labels else []
-        self.value = value
+    # kubelet metrics
+    'kubernetes.kubelet.etcd.cache.entry.count',
+    'kubernetes.kubelet.etcd.cache.hit.count',
+    'kubernetes.kubelet.etcd.cache.miss.count',
+    'kubernetes.kubelet.goroutines',
```

Comment:
We should add some metrics like:
* apiserver_client_certificate_expiration_seconds

And what do you think of:
```text
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.001"} 2
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.002"} 13
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.004"} 18
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.008"} 32
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.016"} 34
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.032"} 34
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.064"} 34
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.128"} 34
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.256"} 34
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="0.512"} 34
rest_client_request_latency_seconds_bucket{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST",le="+Inf"} 34
rest_client_request_latency_seconds_sum{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST"} 0.13816789299999999
rest_client_request_latency_seconds_count{url="http://127.0.0.1:8080/apis/authentication.k8s.io/v1beta1/tokenreviews",verb="POST"} 34
```
This metric could be useful in term of security. 
A spike could indicate some unexpected requests on kubelet API (someone tempting to access to the logs or doing an exec).
But it's hard to parse because the requests in in the label itself (http://127.0.0.1:8080 is the endpoint of the kube-apiserver)

Reply:
The REST counter could be also valuable, to identify any overwhelming kubelet:
```text
# TYPE rest_client_requests_total counter
rest_client_requests_total{code="200",host="127.0.0.1:8080",method="DELETE"} 2
rest_client_requests_total{code="200",host="127.0.0.1:8080",method="GET"} 506
rest_client_requests_total{code="200",host="127.0.0.1:8080",method="PATCH"} 173
rest_client_requests_total{code="200",host="127.0.0.1:8080",method="PUT"} 80
rest_client_requests_total{code="201",host="127.0.0.1:8080",method="POST"} 309
rest_client_requests_total{code="404",host="127.0.0.1:8080",method="GET"} 1
rest_client_requests_total{code="409",host="127.0.0.1:8080",method="POST"} 1
```

Reply:
The following metrics looks like being interesting:
```text
# HELP kubelet_runtime_operations Cumulative number of runtime operations by operation type.
# TYPE kubelet_runtime_operations counter
kubelet_runtime_operations{operation_type="container_status"} 156
kubelet_runtime_operations{operation_type="create_container"} 33
kubelet_runtime_operations{operation_type="exec"} 16
kubelet_runtime_operations{operation_type="exec_sync"} 233
kubelet_runtime_operations{operation_type="image_status"} 73
kubelet_runtime_operations{operation_type="list_containers"} 3800
kubelet_runtime_operations{operation_type="list_images"} 278
kubelet_runtime_operations{operation_type="list_podsandbox"} 3766
kubelet_runtime_operations{operation_type="podsandbox_status"} 146
kubelet_runtime_operations{operation_type="pull_image"} 5
kubelet_runtime_operations{operation_type="remove_container"} 9
kubelet_runtime_operations{operation_type="run_podsandbox"} 28
kubelet_runtime_operations{operation_type="start_container"} 33
kubelet_runtime_operations{operation_type="status"} 398
kubelet_runtime_operations{operation_type="stop_container"} 5
kubelet_runtime_operations{operation_type="stop_podsandbox"} 42
kubelet_runtime_operations{operation_type="version"} 194
```